plugin/nfs/nfs.py

```diff
@@ -0,0 +1,388 @@
+            try:
+                st = os.statvfs(p)
+                total_size = st.f_blocks * st.f_frsize
+                avail_size = st.f_bavail * st.f_frsize
+                fsid = get_fsid(p)
+                fss.append(
+                    FileSystem(fsid, p, total_size,
+                               avail_size, 'none', self._SYSID))
+            except OSError:
+                pass
+
+        return search_property(fss, search_key, search_value)
+
+    def exports(self, search_key=None, search_value=None, flags=0):
+        """List existing exports"""
+        exports = []
+
+        if os.path.exists(NFSPlugin._EXPORTS):
+            with open(NFSPlugin._EXPORTS, "r") as efile:
+                for line in efile:
+                    ret = NFSPlugin._parse_export(shlex.split(line, '#'))
+                    if ret is None:
+                        continue
+                    exports.append(NFSPlugin._make_object(ret[0], ret[1],
+                                                          ret[2]))
+            efile.close()
+
+        return search_property(exports, search_key, search_value)
+
+    def export_fs(self, fs_id, export_path, root_list, rw_list, ro_list,
+                  anon_uid=NfsExport.ANON_UID_GID_NA,
+                  anon_gid=NfsExport.ANON_UID_GID_NA,
+                  auth_type=None, options=None, flags=None):
+        """Add an export"""
+
+        if fs_id is None and export_path is None:
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           'Must provide fs_id or export_path')
+
+        common_opts = []
+        if options is not None:
+            common_opts.extend(options.split(','))
+
+        if anon_uid is not None and anon_uid != NfsExport.ANON_UID_GID_NA:
+            common_opts.append('anonuid=%d' % anon_uid)
+
+        if anon_gid is not None and anon_gid != NfsExport.ANON_UID_GID_NA:
+            common_opts.append('anongid=%d' % anon_gid)
+
+        if auth_type is not None:
+            common_opts.append('sec=%s' % str(auth_type))
+
+        if export_path is None:
+            export_path = NFSPlugin._get_fsid_path(fs_id)
+            if export_path is None:
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                               'Could not locate filesystem')
+
+        for host in rw_list:
+            opts = copy.copy(common_opts)
+            if host in root_list:
+                opts.append('no_root_squash')
+            opts.append('rw')
+            try:
+                NFSPlugin._add_export(export_path, host, opts)
+            except Exception:
+                raise
+
+        for host in ro_list:
+            opts = copy.copy(common_opts)
+            if host in root_list:
+                opts.append('no_root_squash')
+            opts.append('ro')
+            try:
+                NFSPlugin._add_export(export_path, host, opts)
+            except Exception:
+                raise
+
+        NFSPlugin._update_exports()
```

Comment:
It is difficult to know what to do with this, clearly the plugin input here was based on the requirements of certain appliances and their APIs, so i dont think its a good idea to change that.

The output side is a big unknown, nothing in the repo really consumes the output of exports_fs, there is one place in the python bindings that converts it back to a C object, but thats it, lsmcli ignores it, and we have no other examples. so changing the output to a list of objects would be easy enough to do, but the consequences of doing so are not clear to me.

I suppose it would be possible to emulate the behaviour of the appliances and group it so that all hosts for one path are together, but i just feel this would be fragile as the actual behaviour is path-host pairs, and that is what tools like exportfs break it down into.